gluecodium/src/main/resources/templates/ffi/FfiInstanceCacheImpl.mustache

```diff
@@ -49,23 +49,27 @@ void*
     return reinterpret_cast<std::shared_ptr<void>*>(handle)->get();
 }
 
-// "Private" function, call only when mutex is locked.
 void
-{{libraryName}}_process_deletion_queue(int32_t isolate_id) {
-    auto iter = _deletion_queue.find(isolate_id);
-    if (iter == _deletion_queue.end()) return;
+{{libraryName}}_cache_dart_handle_by_raw_pointer(void* raw_pointer, int32_t isolate_id, Dart_Handle dart_handle) {
+    auto new_handle = Dart_NewWeakPersistentHandle_DL(dart_handle, NULL, 0, &{{libraryName}}_nop_finalizer);
+    std::vector<Dart_WeakPersistentHandle> pending_for_deletion;
 
-    for (const auto& weak_handle: iter->second) {
-        Dart_DeleteWeakPersistentHandle_DL(weak_handle);
+    {
+        const std::lock_guard<std::mutex> lock(_instance_cache_mutex);
+        _instance_cache[raw_pointer] = new_handle;
+
+        // Copy deletion queue
+        auto iter = _deletion_queue.find(isolate_id);
+        if (iter == _deletion_queue.end()) return;
+
+        pending_for_deletion = iter->second;
```

Comment:
std swap or std move is better

Reply:
Std swap here should be more efficient. But most efficient is to extract all block to separate function and use std move inside to avoid unnecessary construction of vector.

Reply:
I like the "function" approach the best, as it makes uses of RVO and thus does not need an explicit move.

Reply:
Ah, no, still need a move here. But not below.

Reply:
After some experimentation, `std::swap()` has proven to be more suitable for the vector. Preserving an empty vector in the cache, for reuse, fits the use case very well.